pilot/pkg/model/push_context.go

```diff
@@ -196,6 +196,13 @@ type XDSUpdater interface {
 	ProxyUpdate(clusterID, ip string)
 }
 
+var _ XDSUpdater = &XDSUpdaterHolder{}
+
+// XDSUpdaterHolder is a wrapper that holds an XDSUpdater and implements its interface
+type XDSUpdaterHolder struct {
+	XDSUpdater
+}
```

Comment:
This is confusing to me. So we have an XDSUpdaterHolder which consists of an XDSUpdater, and also IS an XDSUpdater. Then we instantiate a XDSUpdaterHolder with no XDSUpdater specified (eg `nil`). So if we call `xdsUpdater.something()` we are basically calling `nil.something()`? Is that right?

If so why do we need `XDSUpdaterHolder`, can we just use `nil`?

Reply:
Fixed. I've moved the creation of the discovery server to the beginning so that it's always available and we no longer have the chicken or the egg problem.